### PR TITLE
Add --provider option

### LINKED ISSUE #171: Add --provider CLI option
Add a `-p|--provider <provider>` option so the user can choose the provider. This would override the provider list in the answerfile.

### PR DESCRIPTION
**Pull request by aweiteka**
The --provider option is constrained to supported providers.
Providers can be specified in answers.conf. If --provider opt
is specified it will be overridden. Otherwise the default provider
is used.

Resolves #171

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+Providers may be specified using the `answers.conf` file or the `--provider <provider>` option. If not specified the kubernetes provider will be used by default.
+
+Sample `answers.conf` file specifying provider
+
+```
+[general]
+provider = openshift
+```
+
+Using the `--provider <provider>` option will  will override the provider in the answerfile.
```
Comment (navidshaikh) [inline docs/cli.md]: * Using the `--provider <provider>` option will  override the provider in the answerfile.

(multiple 'will' in the line)

File: docs/cli.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Atomica App Command Line Interface (CLI)
```
Comment (dustymabe) [inline docs/cli.md]: s/Atomica/Atomic

**Comment by goern** [comment]
one typo but rest LGTM

**Comment by dustymabe** [comment]
I haven't yet mastered the code base but my naive eyes seem to be ok with everything. LGTM

**Comment by navidshaikh** [comment]
How about adding a test for this? We have multiple examples in `tests/cached_nulecules` - we can use the existing artifacts to test this.

**Comment by goern** [comment]
What do you want to test? 
1. no provider given at all -> default is ...
2. provider given in answers.conf -> its used
3. no provider in answers.conf but on cmd line -> its used
4. provider in answers.conf and on cmd line -> cmd line provided is used

So you just test the behavior of cmd line args and answers.conf?

**Comment by navidshaikh** [comment]
@goern : Correct.
Preferably 
 - new command line option introduced for `provider`
 - the override behavior of `provider`

**Comment by vpavlin** [comment]
Appart from the type - LGTM

**Comment by vpavlin** [comment]
typo, even

**Comment by goern** [comment]
:) #loveit

On 09/01/2015 12:05 PM, Vaclav Pavlin wrote:
> typo, even
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projectatomic/atomicapp/pull/234#issuecomment-136661128>.
> 

-- 
Principal Software Engineer - Systems Design & Engineering
Mobile: +49 171 2801345

Follow Us: https://twitter.com/RedHatRefArch
Plus Us: https://plus.google.com/u/0/b/114152126783830728030/
Like Us: https://www.facebook.com/rhrefarch

Red Hat GmbH, http://www.de.redhat.com/ Sitz: Grasbrunn,
Handelsregister: Amtsgericht München, HRB 153243
Geschäftsführer: Charles Cachera, Michael Cunningham, Michael O'Neill,
Charles Peters

**Comment by charliedrage** [comment]
When you do
```
atomicapp run --help
```
it shows the provider option

but when you do 
```
positional arguments:
  {run,install,stop}

optional arguments:
  -h, --help            show this help message and exit
  -V, --version         show the version and exit.
  -v, --verbose         Verbose output mode.
  -q, --quiet           Quiet output mode.
  --dry-run             Don't actually call provider. The commands that should
                        be run will be sent to stdout but not run.
  -a ANSWERS, --answers ANSWERS
                        Path to answers.conf
  --answers-format ANSWERS_FORMAT
                        The format for the answers.conf.sample file.Default is
                        'ini', Valid formats are 'ini', 'json', 'xml', 'yaml'.
```

It doesnt. Are we able to add it to the default --help on `atomicapp`? makes it seem like a hidden feature :)

**Comment by vpavlin** [comment]
It does not:) That's a common thing to have verb specific options - try it with `git --help`:)

**Comment by charliedrage** [comment]
@vpavlin thought so! thought it'd double check anyways

**Comment by charliedrage** [comment]
otherwise +1 from me!

just tried it with atomicapp run --provider=docker projectatomic/helloapache and it works well.

**Pull request merged**